docs/posts/Computation expressions.fsx

```diff
@@ -0,0 +1,535 @@
+In the case of `yield!`, only the `YieldFrom` method gets called. This means, that the input of type `'U` in the `For` method is actually already a yielded value.
+
+### Math builder
+
+Another thing you can do is to add a `Run` method. This method will be executed at the end of the Computation expression.
+We can also combine this with the ability of the Computation expression to store a state in itself. By this we can control what exactly Run will do:
+
+
+*)
+
+/// Used to specify which mathematical operation should be applied on the float list to reduce it
+type MathOperation = 
+    | Sum
+    | Multiply
+    | Custom of (float list -> float)
+
+    /// Apply the given operation on the float list, resulting in a single float
+    member this.Apply(vals : float list) =
+
+        match this with
+        | Sum -> vals |> List.sum
+        | Multiply -> vals |> List.reduce (fun a b -> a * b)
+        | Custom operation -> operation vals
+
+
+type MathBuilder() = 
+
+    // Mutable math operation state, default value is Sum
+    let mutable operation = Sum
+
+    /// When the user yields a value of type MathOperation, this will not be added to the list of values but overwrite the state of the operation variable
+    member this.Yield(newOperation : MathOperation) =
+        operation <- newOperation
+        []
+
+    /// User can yield ints
+    member this.Yield(v : int) = 
+        printfn $"Yield {v}"
+        [float v]
+
+    /// User can yield floats
+    member this.Yield(v : float) = 
+        printfn $"Yield {v}"
+        [v]
+
+    /// This will combine the yielded values in the background
+    member this.Combine(v1 : float list, v2 : float list) =
+        printfn $"Combine {v1} with {v2}"
+        v1 @ v2
+
+    /// This method is needed. It can be used to lazily evaluate the operations in the CE. Instead here, we just directly execute the given function.
+    member this.Delay(f : unit -> 'T) = 
+        printfn $"Delay"
+        let res = f()
+        printfn $"Delay result: {res}"
+        res
+
+    /// Is called in empty else branches
+    member this.Zero() = 
+        printfn "Zero"
+        []
+
+    /// Allows for using for loops in the computation expression
+    member this.For(vs : 'U seq, f : 'U -> float list) =
+        printfn $"For"
+        vs
+        |> Seq.collect f
+        |> Seq.toList
+
+    /// This will allow for using the yield! method
+    member this.YieldFrom(vs : 'T list) =
+        printfn $"YieldFrom {vs}"
+        vs
+
+    /// This will run the operation on the final float list, returning a single float as output of the compuation expression.
+    member this.Run(vs : float list) =
+        printfn $"Run {vs}"
+        operation.Apply vs
+
+// For easier initialization of the computation expression
+let math = MathBuilder()
+
+(**
+
+So let's test this out
+*)
+
+math {
+    1
+    2
+    3
+}
+
+// Results in 6
+
+let faculty x =
```

Comment:
_faculty_ means _Fakultät_ as an institution or building.
_factorial_ means the mathematical procedure.